spring-kafka/src/main/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainer.java

```diff
@@ -204,14 +206,20 @@ public boolean isContainerPaused() {
 
 	@Override
 	public boolean isChildRunning() {
-		if (!isRunning()) {
-			return false;
-		}
-		for (MessageListenerContainer container : this.containers) {
-			if (container.isRunning()) {
+		this.lifecycleLock.lock();
+		try {
+			for (MessageListenerContainer container : this.containers) {
+				if (container.isRunning()) {
+					return true;
+				}
+			}
+			if (this.startedContainers.get() > 0) {
```

Comment:
How we can reach here if we have at least one running child container?
Otherwise, if we don't, how this property is positive?
Isn't there something off?
Why do we need `this.startedContainers` at all if we fully can control that via ` this.containers`?

Reply:
This is the critical condition and is reachable if the `isChildRunning` is called if the child containers are still processing the messages. `this.containers` are cleared after the `stop` is called. We cannot rely on it completely. `this.startedContainers` could be used to track the actual child count and would be reached to `zero` after all the containers are stopped.

Please find the junit I have submitted with this PR to test this condition. 

```
public void testIsChildRunning() throws Exception {

.....
// This returns true since one container is still processing message. Key validation for this test case.
		assertThat(container.isChildRunning()).isTrue();
```